### ISSUE TITLE
setAge goes to success callback but don't see it in Flurry Dashboard

### ISSUE THREAD

**Issue by gsccheng** [jfpsf/flurry-phonegap-plugin#54]
Has anyone gotten setAge to work recently? I am hitting the success callback, am passing the age as a number integer and appear to have my sessions started and ended properly since I see my other parameters showing in my Flurry dashboard. Yet, in the Audience section of Flurry, it says I have no age data set. Anyone one have any ideas on why this is going on? Thanks.

**Comment by jfpsf** [comment]
How long have you given it?

On Tue, Nov 10, 2015 at 3:25 PM, Gabriel Cheng <notifications@github.com>
wrote:

> Has anyone gotten setAge to work recently? I am hitting the success
> callback, am passing the age as a number integer and appear to have my
> sessions started and ended properly since I see my other parameters showing
> in my Flurry dashboard. Yet, in the Audience section of Flurry, it says I
> have no age data set. Anyone one have any ideas on why this is going on?
> Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jfpsf/flurry-phonegap-plugin/issues/54>.
>



-- 
+++++++++++++++++++++++++++++++++++
John Potter
(415) 846-8021
+++++++++++++++++++++++++++++++++++

**Comment by gsccheng** [comment]
I have been getting the success callback since Friday of last week. My other parameters are showing up (although kind of sporadically). Thanks for your reply.

**Comment by gsccheng** [comment]
I just got a response from Flurry Support that the age would need to be set before the start session call. I'm guessing this would be the case for setting the user id and gender as well. I also saw another [repo](https://github.com/blakgeek/cordova-plugin-flurryanalytics) where these user settings are initialization options that are sent with the start session call. I'm hoping in a few hours I can see my changes.

**Comment by jfpsf** [comment]
Thanks. Let us know, and we will make the change.

On Wed, Nov 11, 2015 at 10:07 AM, Gabriel Cheng <notifications@github.com>
wrote:

> I just got a response from Flurry Support that the age would need to be
> set before the start session call. I'm guessing this would be the case for
> setting the user id and gender as well. I also saw another repo
> <https://github.com/blakgeek/cordova-plugin-flurryanalytics> where these
> user settings are initialization options that are sent with the start
> session call. I'm hoping in a few hours I can see my changes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jfpsf/flurry-phonegap-plugin/issues/54#issuecomment-155863880>
> .
>



-- 
+++++++++++++++++++++++++++++++++++
John Potter
(415) 846-8021
+++++++++++++++++++++++++++++++++++

**Comment by gsccheng** [comment]
I still don't see it in my dashboard. Flurry support is now saying that you can call it before or after starting the session-- the difference is calling it before will have the age included in the start of session report.

I'm looking at my Xcode log:
```
2015-11-11 10:20:35.067 App[12034:90570] Setting Flurry version to 1.0.9
2015-11-11 10:20:35.067 App[12034:90457] Enabling Flurry Event Logging
2015-11-11 10:20:35.076 App[12034:90437] Setting Flurry User ID to 3
2015-11-11 10:20:35.076 App[12034:90568] Setting Flurry Age to 30
2015-11-11 10:20:35.092 App[12034:90570] Flurry: setAppVersion[1.0.9]
2015-11-11 10:20:35.098 App[12034:90437] Flurry: setUserID[3]
...
```
I think the first 4 lines are from the plugin and the last 2 from the Flurry SDK. I am able to see my app version and user ID in my Flurry dashboard. So, notice how you see the two `Flurry:` lines but the equivalent age output is nowhere to be found. Perhaps this has something to do with it? I would expect something like `Flurry: setAge[30]`. Perhaps something in the plugin isn't setting up the Flurry method call properly?

Below is the continuation of the log for the start session report in the Flurry debug log in Xcode.
```
...
2015-11-11 10:20:35.187 App[12034:87663] Flurry: Background session enabled[0]
2015-11-11 10:20:35.189 App[12034:87663] Flurry: setUserID[3]
2015-11-11 10:20:35.189 App[12034:90437] FlurrySession: Set userID[3]
2015-11-11 10:20:35.191 App[12034:90437] FlurrySession: Add session with startTime[2015-11-11 17:49:17 +0000] to saved sessions
2015-11-11 10:20:35.194 App[12034:90437] FlurrySession: Add crashed former session
2015-11-11 10:20:35.195 App[12034:90437] FlurrySession: Event logging enabled[1]
2015-11-11 10:20:35.195 App[12034:90437] FlurrySession: Session reports on close enabled[1]
2015-11-11 10:20:35.196 App[12034:90437] FlurrySession: Session reports on pause enabled[0]
2015-11-11 10:20:35.196 App[12034:90437] FlurrySession: Crash reporting enabled[0]
2015-11-11 10:20:35.199 App[12034:90437] FlurrySession: Initialized session from scratch with startTime[2015-11-11 18:20:35 +0000]
2015-11-11 10:20:35.200 App[12034:90437] FlurrySession: Created active session with API[KEY***]
2015-11-11 10:20:35.202 App[12034:90437] FlurrySession: dealloc session
2015-11-11 10:20:35.221 App[12034:90437] FlurrySession: Session reports on close enabled[1]
2015-11-11 10:20:35.223 App[12034:90437] FlurrySession: Session reports on pause enabled[1]
2015-11-11 10:20:35.224 App[12034:90437] FlurrySession: Event logging enabled[1]
2015-11-11 10:20:35.224 App[12034:90437] FlurrySession: Crash reporting enabled[1]
2015-11-11 10:20:35.229 App[12034:90437] FlurrySession: set age to [1985-01-01 08:00:00 +0000]
2015-11-11 10:20:35.229 App[12034:90437] FlurrySession: Set userID[3]
```

However, here in the second to last line you can see Flurry with `1985-01-01 08:00:00 +0000`. So, it looks like it did receive the data of age 30 but somehow it is converted to a timestamp. I'm also wondering if that is the cause of my problem as well.

If anyone has any insight into this, I would love to hear it. Thanks!

**Comment by gsccheng** [comment]
I'm going to close this issue. I see the age on the dashboard now. I actually tried 4 different cases setting the age either before or after starting the session and whether it is a string or number. After multiple days, I see the age for each case show up other than when it is a string and set after. Thanks!

**Comment by jfpsf** [comment]
Thanks. Flurry is very random as to when the data shows up.

On Sun, Nov 15, 2015 at 12:25 PM, Gabriel Cheng <notifications@github.com>
wrote:

> Closed #54 <https://github.com/jfpsf/flurry-phonegap-plugin/issues/54>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jfpsf/flurry-phonegap-plugin/issues/54#event-464869583>
> .
>



-- 
+++++++++++++++++++++++++++++++++++
John Potter
(415) 846-8021
+++++++++++++++++++++++++++++++++++
